src/shims/foreign_items.rs

```diff
@@ -167,6 +170,28 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         };
 
         // Next: functions that return.
+        if this.emulate_foreign_item_by_name(link_name, args, dest, ret)? {
+            this.dump_place(*dest);
+            this.go_to_block(ret);
+        }
+
+        Ok(None)
+    }
+
+    /// Emulates calling a foreign item using its name, failing if the item is not supported.
+    /// Returns Ok(false) if after calling this function, the call should return earlier instead of
+    /// going to the next block.
```

Comment:
What does this mean, "the call should return earlier"?

What about: "Returns `true` if the caller is expected to jump to the return block, and `false` if jumping has already been taken care of"